packages/main/README.rst

```diff
@@ -327,6 +327,70 @@ Found a bug? Missing a critical feature? Interested in contributing?
 Head over to the `Contribution guide <https://rpaframework.org/contributing/guide.html>`_
 to see where to get started.
 
+Development
+-----------
+
+Repository development is **`Python <https://www.python.org/>_`** based and requires at minimum
+Python version 3.7+ installed on the development machine. The default Python version used in the
+Robocorp Robot template is 3.7.5 so it is a good choice for the version to install. Not recommended
+versions are 3.7.6 and 3.8.1, because they have issues with some of the dependencies related to rpaframework.
+At the time the newer Python versions starting from 3.9 are also not recommended, because some of
+the dependencies might cause issues.
+
+Repository development tooling is based on basically on `poetry`_ and `invoke`_. Poetry is the
+underlying tool used for compiling, building and running the package. Invoke is used for scripting
+purposes for example for linting, testing and publishing tasks.
+
+First steps to start developing:
+
+#. clone the repository
+#. create a new Git branch or switch to correct branch or stay in master branch
+#. create the .venv directory into package you are developing
+#. poetry install
+#. if testing against Robocorp Robot which is using devdata/env.json
+   - set environment variables
+   - or poetry build and use resulting .whl file in the Robot conda.yaml
+   - or poetry build and push resulting .whl file into a repository and use raw url
+     to include it in the Robot conda.yaml
+#. poetry run python -m robot <ROBOT_ARGS> <TARGET_ROBOT_FILE>
+#. poetry run python <TARGET_PYTHON_FILE>
+#. invoke lint
+#. invoke test (this will run both Python unittests and robotframebased tests defined in the
+   packages tests/ directory)
+#. Git commit changes
+#. Git push changes to remote
+#. create pull request from the branch describing changes included in the description
+#. update docs/source/releasenotes.rst with changes (commit and push)
+
+Packaging and publishing are done after changes have been merged into master branch.
+All the following steps should be done within master branch.
+
+#. in the package directory containing changes execute invoke lint and invoke test
+#. update pyproject.toml with new version according to semantic versioning
+#. update docs/source/releasenotes.rst with changes
+#. in the repository root (so called "meta" package level) run command poetry update
```

Comment:
Nice, this goes perfectly along with the change I'm going to commit with having all meta's local deps inherited from local packages (instead of PyPI, as it would require to publish the new versions first).